### ISSUE TITLE
Common error on importing existing features

### ISSUE THREAD

**Issue by Jdgilby** [Riverscapes/QRiS#319]
This happens when trying to import existing feature class through right click on layer. This is the most common bug I'm running into when importing existing data.
![image](https://github.com/Riverscapes/QRiS/assets/97305413/7b0b9cb0-1894-47b5-af6d-9ad1bc21fb86)
I've tried a few data cleanup functions to the importing layer without success.

This is on the active extents layer, if that matters at all.

**Comment by KellyMWhitehead** [comment]
TLDR: I just make a fix that should address this error for 0.3.7.  

This was tricky to figure out. I eventually found that there were invalid geometries in the original dataset (QGiS and even OGR were having trouble identifying these as invalid). For the dataset you provided above, if you try to zoom to feature from the attribute table (the first feature, based on the message above), it should show up as invalid. 

In the dataset I was looking at, the geometry was of type POLYGON, but had 2 parts, which should only be allowed for MULTIPOLYGON types. It wasn't until I iterated through these parts when OGR detected that they were invalid. QRIS will now skip any geometires or geometry parts that are null, or invalid (after trying to repair them first).

**Comment by Jdgilby** [comment]
@KellyMWhitehead This is working now, but should there be some sort of prompt to let users know a feature was skipped/failed to import due to bad geometry?

**Comment by Jdgilby** [comment]
Might be a good idea. Thinking of less experienced users. Might wonder why some features import but others wont.

**Comment by KellyMWhitehead** [comment]
Changed severity to warning if features were skipped (or created from multipart features) and success (green check) if all features imported without changes.
